src/foundry/foundry.js/applications/formApplications/documentSheet.d.ts

```diff
@@ -1,106 +1,123 @@
-/**
- * Extend the FormApplication pattern to incorporate specific logic for viewing or editing Document instances.
- * See the FormApplication documentation for more complete description of this interface.
- * @typeParam P - the type of the options object
- * @typeParam D - The data structure used to render the handlebars template.
- * @typeParam O - the type of the Document which should be managed by this form sheet
- */
-declare abstract class DocumentSheet<
-  P extends DocumentSheet.Options = DocumentSheet.Options,
-  D extends object = DocumentSheet.Data<Entity>,
-  O extends Entity<any, any> = D extends DocumentSheet.Data<infer T> ? T : Entity
-> extends FormApplication<P, D, O> {
-  /**
-   * @param object  - An Entity which should be managed by this form sheet.
-   * @param options - Optional configuration parameters for how the form behaves.
-   */
-  constructor(object: O, options?: Partial<P>);
+import { Document } from '../../../common/abstract/module.mjs';
 
+declare global {
   /**
-   * {@inheritdoc}
-   *
-   * @defaultValue
-   * ```typescript
-   * foundry.utils.mergeObject(super.defaultOptions, {
-   *   classes: ['sheet'],
-   *   template: `templates/sheets/${this.name.toLowerCase()}.html`,
-   *   viewPermission: CONST.ENTITY_PERMISSIONS.LIMITED
-   * });
-   * ```
+   * Extend the FormApplication pattern to incorporate specific logic for viewing or editing Document instances.
+   * See the FormApplication documentation for more complete description of this interface.
+   * @typeParam P - the type of the options object
+   * @typeParam D - The data structure used to render the handlebars template.
+   * @typeParam O - the type of the Document which should be managed by this form sheet
    */
-  static get defaultOptions(): DocumentSheet.Options;
+  abstract class DocumentSheet<
+    P extends DocumentSheet.Options = DocumentSheet.Options,
+    D extends DocumentSheet.Data = DocumentSheet.Data,
+    O extends Document<any, any> = D extends DocumentSheet.Data<infer T> ? T : Document<any, any>
+  > extends FormApplication<P, D, O> {
+    /**
+     * @param object  - A Document instance which should be managed by this form.
+     * @param options - Optional configuration parameters for how the form behaves.
+     *                  (default: `{}`)
+     */
+    constructor(object: O, options?: Partial<P>);
```

Comment:
constructor isn't actually overridden in `foundry.js`

Reply:
Where do we put the documentation? In Foundry the constructor documentation is put in the class docs.

Reply:
I guess, just put it there, or leave it out altogether. It’s not like the documentation is different from that of the base class, if I’m not mistaken.